### PR TITLE
feat(Global Footer): add affiliate program link

### PR DESCRIPTION
**Pull request by saghdaey**
### Overview

Add link to Codecademy Affiliate Program to Global Footer

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: EGG-999
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut-labs/src/GlobalFooter/FooterNavLinks/CompanyLinks.tsx [hunk 1]
```diff
@@ -96,6 +96,15 @@ export const CompanyLinks: React.FC<CompanyLinksProps> = ({
             We&apos;re Hiring
           </Anchor>
         </FooterLinkItem>
+        <FooterLinkItem>
+          <Anchor
+            href="https://www.codecademy.com/pages/codecademy-affiliate-program"
+            onClick={(event) => onClick({ event, target: 'affiliate program' })}
```
Comment (saghdaey) [inline packages/gamut-labs/src/GlobalFooter/FooterNavLinks/CompanyLinks.tsx:102]: checking w/ Richie what the copy should be for the "target" here

File: packages/gamut-labs/src/GlobalFooter/FooterNavLinks/CompanyLinks.tsx [hunk 2]
```diff
@@ -96,6 +96,15 @@ export const CompanyLinks: React.FC<CompanyLinksProps> = ({
             We&apos;re Hiring
           </Anchor>
         </FooterLinkItem>
+        <FooterLinkItem>
+          <Anchor
+            href="https://www.codecademy.com/pages/codecademy-affiliate-program"
```
Comment (saghdaey) [inline packages/gamut-labs/src/GlobalFooter/FooterNavLinks/CompanyLinks.tsx:101]: guessing this should actually be` /pages/codecademy-affiliate-program`

Comment (dreamwasp) [reply packages/gamut-labs/src/GlobalFooter/FooterNavLinks/CompanyLinks.tsx]: yaaa

**Review by dreamwasp — APPROVED** [review]
✅

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.52-alpha.715a89.0<br/>@codecademy/gamut-labs@14.4.4-alpha.715a89.0<br/>@codecademy/styleguide@36.4.4-alpha.715a89.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60881c0a40e5b15d0027c212--gamut-preview.netlify.app>https://60881c0a40e5b15d0027c212--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60881c0a40e5b15d0027c212>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.53-alpha.59c87c.0<br/>@codecademy/gamut-labs@14.4.5-alpha.59c87c.0<br/>@codecademy/styleguide@36.5.1-alpha.59c87c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6088521cfb4d7400a25c7d98--gamut-preview.netlify.app>https://6088521cfb4d7400a25c7d98--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6088521cfb4d7400a25c7d98>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.54-alpha.ae57ec.0<br/>@codecademy/gamut-labs@14.4.6-alpha.ae57ec.0<br/>@codecademy/styleguide@36.5.2-alpha.ae57ec.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60886a298d700e009721a1ee--gamut-preview.netlify.app>https://60886a298d700e009721a1ee--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60886a298d700e009721a1ee>Deploy Logs</a></p>

**Pull request merged**